examples/imitation_learning_openbot/policies.py

```diff
@@ -121,23 +121,19 @@ def step(self, obs, position_xy_desired):
         for input_detail in self._input_details:
```

Comment:
Why are we checking for `"cmd_input"` or `"goal_input"` in this if statement? Are both present, and we pass in the same data for each? That seems weird.

```
        for input_detail in self._input_details:
            if "img_input" in input_detail["name"]:

                # select the bottom right corner of obs, normalize, convert to np.float32
                img_input_height = input_detail["shape"][1] 
                img_input_width  = input_detail["shape"][2] 
                obs_height       = obs["camera.final_color"].shape[0]
                obs_width        = obs["camera.final_color"].shape[1]

                assert obs_height - img_input_height >= 0
                assert obs_width - img_input_width   >= 0

                img_input = obs["camera.final_color"][np.newaxis, image_height-target_height:, image_width-target_width:, :-1].astype(np.float32) / 255.0
                self._interpreter.set_tensor(input_detail["index"], img_input)

            elif "cmd_input" in input_detail["name"] or "goal_input" in input_detail["name"]:

                # compute a goal observation
                cm_to_m = 0.01
                
                location_xy_current = obs["location"][0:2] * cm_to_m
                location_xy_error   = np.linalg.norm(location_xy_desired - location_xy_current)
                
                rotation_yaw_current = np.deg2rad(obs["rotation"][1])
                heading_xy_current   = np.array([np.cos(rotation_yaw_current), np.sin(rotation_yaw_current)])
                heading_xy_desired   = (location_xy_desired - location_xy_current) / np.linalg.norm(location_xy_desired - location_xy_current)
                rotation_yaw_error   = np.arctan2(heading_xy_desired[1], heading_xy_desired[0]) - np.arctan2(heading_xy_current[1], heading_xy_current[0])
                
                if rotation_yaw_error < -np.pi:
                    rotation_yaw_error += 2*np.pi
                if rotation_yaw_error > np.pi:
                    rotation_yaw_error -= 2*np.pi

                goal = np.array([location_xy_error, np.sin(rotation_yaw_error), np.cos(rotation_yaw_error)])
                self._interpreter.set_tensor(input_detail["index"], goal[np.newaxis])

            else:
                assert False
```

Reply:
Updated this